src/components/Error.tsx

```diff
@@ -0,0 +1,47 @@
+import React from 'react'
+import { StyleSheet, Text, TouchableOpacity, View } from 'react-native'
+
+interface ErrorDisplayProps {
+  description: string
+  retryConnecting: () => void
+}
+
+export function ErrorDisplay(props: ErrorDisplayProps): React.JSX.Element {
+  return (
+    <View style={styles.container}>
+      <Text style={styles.errorText}>Error: {props.description}</Text>
+      <TouchableOpacity
+        style={styles.retryButton}
+        onPress={props.retryConnecting}>
+        <Text style={styles.retryButtonText}>Try again</Text>
```

Comment:
In future PRs we can use [react-native-localization](https://github.com/stefalda/ReactNativeLocalization), what do you think? It is very used in most of the projects